client/js/options.js

```diff
@@ -98,6 +99,15 @@ settings.on("change", "input, select, textarea", function() {
 			// otherwise, users get notifications for everything
 			return h !== "";
 		});
+		// Construct regex with wordboundary for every highlight item
+		const highlightsTokens = options.highlights.map(function(h) {
+			return escapeRegExp(h);
+		});
+		if (highlightsTokens && highlightsTokens.length) {
+			module.exports.highlightsRE = new RegExp("\\b(?:" + highlightsTokens.join("|") + ")\\b", "i");
```

Comment:
That's a great point, it probably does. I was wondering why `shouldOpenMessagePreview` would not be stored somehow, but it turns out functions are not serialized by `JSON.stringify`:

```js
> JSON.stringify({ foo: 'bar', test() { return 'foo'; }, regex: new RegExp("test") })
'{"foo":"bar","regex":{}}'
```

Kinda sucks that we do `module.exports = options;` but then also use this file to do other things than just dealing with `options` directly.

@xPaw, what do you think?
I can think of a few ideas: wrapping `options` in a property `module.exports.whatever = options;` (requires editing of every place that has `require("[...]/options")`, creating a `optionsHelper.js` file, making `highlightsRE` a function instead as they're not serialized (that would only delay the issue instead of fixing it though), but none of those things are ideal. Open to ideas.